_posts/2023-04-21-tensorflow-mnist-classification.adoc

```diff
@@ -0,0 +1,405 @@
+    servers=KAFKA_SERVERS,
+    stream_timeout=9000,
+    configuration=[
+        "session.timeout.ms=10000",
+        "max.poll.interval.ms=10000",
+        "auto.offset.reset=earliest"
+    ],
+)
+
+manual_ds = manual_ds.map(decode_kafka_stream_record)
+----
+
+If you want to create new stream and verify that our model can provide prediction as the new data arrives, you can easily do so:
+
+[source,bash]
+----
+$ head -5 mnist_test.sql | sed s/test/manual/ > mnist_manual.sql
+$ psql -h localhost -U postgres -f postgres/mnist_manual.sql
+----
+
+In such case you don't need to change Kafka consumer group, but you have to change Kafka topic:
+
+[source,python]
+----
+manual_ds = tfio.experimental.streaming.KafkaGroupIODataset(
+    topics=["tf.public.mnist_manual"],
+    group_id=KAFKA_CONSUMER_GROUP,
+    servers=KAFKA_SERVERS,
+    stream_timeout=9000,
+    configuration=[
+        "session.timeout.ms=10000",
+        "max.poll.interval.ms=10000",
+        "auto.offset.reset=earliest"
+    ],
+)
+
+manual_ds = manual_ds.map(decode_kafka_stream_record)
+----
+
+In either case the result should look like this:
+
+++++
+<div class="imageblock centered-image">
+    <img src="/assets/images/2023-04-21-tensorflow-mnist-classification/tensorflow_mnist_streaming.png" class="responsive-image" alt="TensorFlow digit recognition from streaming">
+</div>
+++++
+
+
+== Conclusions
+
+In this demo we have shown how to load existing data from the database, transform it on the fly, ingest the data into TensorFlow model via Kafka and use the data for model training.
+Later on we ingested newly created data into this pre-trained model using CDC and data streaming and obtained meaningful results.
+We can conclude that Debezium can provide valuable service not only for use cases like the one described in this post, but it can also play a key role in ingesting data to online machine learning pipelines.
+
+While the whole pipeline is not very hard to implement, there are definitely areas which can be improved, make the user experience better and/or make the whole pipeline more smooth.
+As our (Debezium developers) background is not primarily in machine learning and data science, we would appreciate any input from the community, how Debezium can be used in machine learning pipelines (or is already used, if there are any such cases) and where are the rooms for improvements.
+We would also appreciate any new ideas how Debezium, or in general change data capture, can be useful in this area.
+These ideas may reveal further Debezium potential to ingest data into machine learning pipelines and may contribute to better user experience in the whole process.
+In case you have in input any this regard, don't hesitate to reach to use on the http://debezium.zulipchat.com/[Zulip chat], https://groups.google.com/forum/#!forum/debezium[mailing list] or you can transform your ideas directly into http://issues.redhat.com/projects/DBZ/issues[Jira feature requests].
```

Comment:
```suggestion
While the whole pipeline is relatively easy to implement, some areas can be improved to improve the user experience and/or make the entire pipeline more smooth.
As our (Debezium developers) background is not primarily in machine learning and data science, we would appreciate any input from the community on how Debezium can aid machine learning pipelines (or is already used, if there are any such cases) and where are the rooms for improvements.
We would also appreciate any new ideas on how Debezium, or in general, change data capture, can be helpful in this area.
These ideas further reveal Debezium's potential to ingest data into machine learning pipelines and contribute to better user experience in the whole process.
In case you have any input any this regard, don't hesitate to reach out to us on the http://debezium.zulipchat.com/[Zulip chat], https://groups.google.com/forum/#!forum/debezium[mailing list] or you can transform your ideas directly into http://issues.redhat.com/projects/DBZ/issues[Jira feature requests].
```